test/dynamo/test_higher_order_ops.py

```diff
@@ -296,6 +296,52 @@ def f(x):
         arg_count = ifdynstaticdefault(3, 4)
         self._test_wrap_simple(f, default_args_generator((x,)), arg_count)
 
+    def test_allow_python_side_effects_utility(self):
+        from torch._dynamo.utils import _UNSAFE_allow_side_effects
```

Comment:
Since this is your API, let's rename it to something that reflects what it does more. My understand is that this function "disables the side effect safety checks for the current subtracer for the currently executing function".

Maybe something like... _disable_side_effect_safety_checks_for_current_subtracer ? :P